src/nginx/grpc_server_call.cc

```diff
@@ -341,6 +345,26 @@ void NgxEspGrpcServerCall::OnDownstreamWriteable(ngx_http_request_t *r) {
   }
 }
 
+void NgxEspGrpcServerCall::OnHttpBlockReading(ngx_http_request_t *r) {
+  ngx_log_debug0(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
+                 "NgxEspGrpcServerCall::OnHttpBlockReading");
+
+  if (r->connection->error){
+    ngx_log_debug0(NGX_LOG_DEBUG_HTTP, r->connection->log, 0,
+                   "NgxEspGrpcServerCall::OnHttpBlockReading: connection error");
+
+    ngx_esp_request_ctx_t *ctx = ngx_http_esp_ensure_module_ctx(r);
+    NgxEspGrpcServerCall *server_call = ctx->grpc_server_call;
```

Comment:
can we set ctx->grpc_server_call to nullptr here
https://github.com/cloudendpoints/esp/blob/master/src/nginx/module.cc#L190

and here check if it is nullptr.

I am worrying that this is called after "r" is gone. which mat cause esp to crash

Reply:
As discussed offline, setting `grpc_server_call` to nullptr in the destructor. 
Added a nullptr check here.